src/cli/dispatchers/messages.py

```
@@ -322,37 +329,41 @@ async def _update_token_tracking(self, node_data: dict[str, Any]) -> None:
+                    agent_config.compression.auto_compress_threshold,
                 )
-
-                with console.console.status(
-                    f"[{theme.spinner_color}]Context at {usage_pct}%, auto-compressing to new thread...[/{theme.spinner_color}]"
-                ):
-                    compression_handler = CompressionHandler(self.session)
-                    await compression_handler.handle()
+            ):
+                self._pending_compression = True
 
         except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)